frontend/src/components/common/form/FastInput.tsx

```diff
@@ -90,8 +95,14 @@ export const FastInput: React.FC<FastInputProps> = memo(
           isValid={!!touch && !error && value && !disabled}
           value={value}
           placeholder={placeholder}
-          onBlur={handleBlur}
+          onBlur={(e: any) => {
+            if (onBlurFormatter) {
```

Comment:
This allows individual fields to be responsible for cleaning up their own data, instead of making the form to the cleanup during onSubmit.